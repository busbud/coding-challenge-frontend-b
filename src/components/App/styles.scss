
@import '../../shared-styles/variables';

.header {
  text-align: center;
}

.nav {
  width: $content-width;
  margin: 0 auto;
  padding: $padding;
  text-align: center;
}

.content {
  width: $content-width;
  margin: $padding auto;
}

.navItem {
  display: inline-block;
}

.link {
  padding: $padding;
  color: #00a68f;
  font-size: 18px;
}

.logo {
  display: block;
  width: 252px;
  height: 252px;
  margin: $padding auto;
  background: url('../../images/icon-kytLogo_large-252x252.svg');
}
