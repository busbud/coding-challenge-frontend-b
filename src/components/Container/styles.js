import styled from 'styled-components'

export const StyledContainer = styled.div`
  padding: 0 10px;
  margin: 0 auto;
  max-width: 620px;
  width: 100%;
`
