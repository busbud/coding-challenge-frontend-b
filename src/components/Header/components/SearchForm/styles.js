import styled from "styled-components";

import { BREAKPOINTS } from "utils/constants";

export const Form = styled.div`
  background-color: var(--white);
  border-radius: 4px;
  margin-top: 16px;
  padding: 16px;

  @media screen and (min-width: ${BREAKPOINTS.tablet}) {
    align-items: center;
    display: flex;
    padding: 12px 16px;
  }
`;

export const FormControl = styled.div`
  &:not(:last-of-type) {
    margin-bottom: 16px;
  }

  @media screen and (min-width: ${BREAKPOINTS.tablet}) {
    flex: 1;
    margin-bottom: 0;

    &:not(:last-of-type) {
      margin-right: 16px;
      margin-bottom: 0;
    }
  }
`;
