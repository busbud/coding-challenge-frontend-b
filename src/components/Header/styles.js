import styled from "styled-components";

export const StyledHeader = styled.header`
  background-color: var(--blue);
  color: var(--white);
  text-align: left;
  padding: 32px 0;
  margin-bottom: 24px;
`;

export const Top = styled.div`
  width: 100%;
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin-bottom: 40px;

  .ant-select-arrow {
    color: var(--white);
  }
`;

export const Logo = styled.a`
  width: 108px;
`;

export const Title = styled.h2`
  color: var(--white);
  font-size: 36px;
  line-height: 40px;
  margin-bottom: 16px;
`;
