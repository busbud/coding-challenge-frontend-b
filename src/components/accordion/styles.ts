// Packages
import styled, { css } from 'styled-components'

import * as L from 'layout'

export const ContentWrapper = styled.div<{ maxHeight: string | undefined }>`
  ${({ theme, maxHeight }) => css`
    background-color: ${theme.colors.white};
    overflow: hidden;
    transition: ${theme.transition.ease};
    max-height: ${maxHeight || 0};

    & > * {
      transition: ${theme.transition.ease};
      opacity: ${maxHeight ? 1 : 0};
    }
  `}
`

export const TriggerWrapper = styled(L.Cluster)`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    transition: ${theme.transition.ease};
  `}
`

export const Wrapper = styled(L.Box)<{ withBorder?: boolean }>`
  ${({ theme, withBorder }) => css`
    border: ${withBorder && `1px solid ${theme.colors.muted}`};
    padding: 1rem;
    cursor: pointer;
    box-sizing: border-box;
    box-shadow: ${theme.shadow.small};
  `}
`
