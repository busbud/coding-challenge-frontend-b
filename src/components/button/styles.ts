import styled, { css } from 'styled-components'
import media from 'styled-media-query'
import { darken } from 'polished'

type Props = {
  rounded?: boolean
  circle?: boolean
  primary?: boolean
  secondary?: boolean
  skyBlue?: boolean
  disabled?: boolean
}

const modifiers = {
  rounded: (theme) => css`
    border-radius: ${theme.border.rounded};
    ${media.greaterThan('medium')`
      padding: 1.3rem ${theme.spaces.large};
    `}
  `,
  circle: (theme) => css`
    border-radius: ${theme.border.circle};
    width: 32px;
    height: 32px;
    padding: 0;
    display: flex;
    align-items: center;
    justify-content: center;
  `,
  primary: (theme) => css`
    background-color: ${theme.colors.primary};
    &:hover {
      background: ${darken(0.1, theme.colors.primary)};
    }
  `,
  secondary: (theme) => css`
    background-color: ${theme.colors.secondary};
    &:hover {
      background: ${darken(0.1, theme.colors.secondary)};
    }
  `,
  skyBlue: (theme) => css`
    background-color: ${theme.colors.blue};
    &:hover {
      background: ${darken(0.1, theme.colors.blue)};
    }
  `,
  disabled: (theme) => css`
    background-color: ${theme.colors.muted};
    cursor: not-allowed;
    &:hover {
      background: ${darken(0.1, theme.colors.muted)};
    }
  `
}

export const ButtonWrapper = styled.button<Props>`
  ${({ theme, rounded, circle, primary, secondary, skyBlue, disabled }) => css`
    background-color: ${theme.colors.gray};
    border-radius: ${theme.border.radius};
    color: ${theme.colors.white};
    padding: 1rem ${theme.spaces.small};
    font-size: ${theme.font.sizes.base};
    text-align: center;
    white-space: nowrap;
    cursor: pointer;
    transition: ${theme.transition.ease};

    ${rounded && modifiers.rounded(theme)};
    ${circle && modifiers.circle(theme)};
    ${primary && modifiers.primary(theme)};
    ${secondary && modifiers.secondary(theme)};
    ${skyBlue && modifiers.skyBlue(theme)};
    ${disabled && modifiers.disabled(theme)};
  `}
`
