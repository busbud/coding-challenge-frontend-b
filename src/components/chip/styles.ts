import styled, { css } from 'styled-components'

export const Card = styled.article`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    padding: ${theme.spaces.medium};
    color: ${theme.colors.texts};
    font-size: ${theme.font.sizes.xsmall};
    border-radius: ${theme.border.radius};
  `}
`

export const TitleWrapper = styled.div`
  ${({ theme }) => css`
    border-left: 6px solid ${theme.colors.primary};
    padding-left: ${theme.spaces.xsmall};
    margin-bottom: ${theme.spaces.xsmall};
  `}
`

export const Title = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    font-weight: bold;
  `}
`

export const SubTitle = styled.h4`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xxsmall};
    font-weight: normal;
    color: ${theme.colors.gray};
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    p + p {
      margin-top: ${theme.spaces.small};
    }
  `}
`
