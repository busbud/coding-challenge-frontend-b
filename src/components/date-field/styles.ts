// Packages
import styled, { css } from 'styled-components'
import { rgba } from 'polished'

export const Overlay = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
    background: ${theme.colors.white};
    box-shadow: ${rgba(theme.colors.muted, 1)};
    border: 2px solid ${rgba(theme.colors.gray, 0.3)};
    border-radius: ${theme.border.radius};

    position: absolute;
    transform: translateY(4px);
    z-index: 100;
  `}
`

export const OverlayToolbar = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    display: flex;
    flex-direction: row;
    padding-left: ${theme.spaces.xsmall};
    padding-right: ${theme.spaces.xsmall};
    padding-top: ${theme.spaces.xsmall};
  `}
`

export const LibStyles = styled.div`
  ${({ theme }) => css`
    .DayPicker {
      display: inline-block;
      font-size: ${theme.font.sizes.small};
      margin-top: 0.4rem;
    }

    .DayPicker-wrapper {
      position: relative;
      flex-direction: row;
      padding-bottom: 1rem;

      -webkit-user-select: none;
      -moz-user-select: none;
      -ms-user-select: none;

      user-select: none;
    }

    .DayPicker-Months {
      display: flex;
      flex-wrap: wrap;
      justify-content: center;
    }

    .DayPicker-Month {
      display: table;
      margin: 0 1rem;
      margin-top: 1rem;
      border-spacing: 0;
      border-collapse: collapse;

      -webkit-user-select: none;
      -moz-user-select: none;
      -ms-user-select: none;

      user-select: none;
    }

    .DayPicker-NavButton {
      position: absolute;
      top: 1rem;
      right: 1.5rem;
      left: auto;

      display: inline-block;
      margin-top: 2px;
      width: 1.25rem;
      height: 1.25rem;
      background-position: center;
      background-size: 50%;
      background-repeat: no-repeat;
      color: ${theme.colors.primary};
      cursor: pointer;
    }

    .DayPicker-NavButton:hover {
      opacity: 0.8;
    }

    .DayPicker-NavButton--prev {
      margin-right: 1.5rem;
      border-right: 2px solid ${rgba(theme.colors.gray, 0.3)};
      border-bottom: 2px solid ${rgba(theme.colors.gray, 0.3)};
      height: 12px;
      width: 12px;
      transform: rotate(130deg);
    }

    .DayPicker-NavButton--next {
      border-right: 2px solid ${rgba(theme.colors.gray, 0.3)};
      border-bottom: 2px solid ${rgba(theme.colors.gray, 0.3)};
      height: 12px;
      width: 12px;
      transform: rotate(-45deg);
    }

    .DayPicker-NavButton--interactionDisabled {
      display: none;
    }

    .DayPicker-Caption {
      display: table-caption;
      margin-bottom: 0.5rem;
      padding: 0 0.5rem;
      text-align: left;
      font-size: ${theme.font.sizes.small};
    }

    .DayPicker-Caption > div {
      font-weight: 500;
      font-size: ${theme.font.sizes.small};
    }

    .DayPicker-Weekdays {
      display: table-header-group;
      margin-top: 1rem;
    }

    .DayPicker-WeekdaysRow {
      display: table-row;
    }

    .DayPicker-Weekday {
      display: table-cell;
      padding: 0.5rem;
      color: ${theme.colors.gray};
      text-align: center;
      font-size: ${theme.font.sizes.small};
    }

    .DayPicker-Weekday abbr[title] {
      border-bottom: none;
      text-decoration: none;
    }

    .DayPicker-Body {
      display: table-row-group;
    }

    .DayPicker-Week {
      display: table-row;
    }

    .DayPicker-Day {
      display: table-cell;
      padding: 0.5rem;
      border-radius: 50%;
      vertical-align: middle;
      text-align: center;
      cursor: pointer;
    }

    .DayPicker-WeekNumber {
      display: table-cell;
      padding: 0.5rem;
      min-width: 1rem;
      border-right: 1px solid ${theme.colors.muted};
      color: ${theme.colors.gray};
      vertical-align: middle;
      text-align: right;
      font-size: ${theme.font.sizes.small};
      cursor: pointer;
    }

    .DayPicker--interactionDisabled .DayPicker-Day {
      cursor: default;
    }

    .DayPicker-Footer {
      padding-top: 0.5rem;
    }

    .DayPicker-TodayButton {
      border: none;
      background-color: transparent;
      background-image: none;
      box-shadow: none;
      color: ${theme.colors.blue};
      font-size: ${theme.font.sizes.small};
      cursor: pointer;
    }

    .DayPicker-Day--today {
      color: ${theme.colors.blue};
      font-weight: 700;
    }

    .DayPicker-Day--outside {
      color: ${theme.colors.gray};
      cursor: default;
    }

    .DayPicker-Day--disabled {
      color: ${rgba(theme.colors.gray, 0.7)};
      cursor: default;
    }

    .DayPicker-Day--sunday {
      background-color: #f7f8f8;
    }

    .DayPicker-Day--sunday:not(.DayPicker-Day--today) {
      color: #dce0e0;
    }

    .DayPicker:not(.DayPicker--interactionDisabled)
      .DayPicker-Day:not(.DayPicker-Day--disabled):not(.DayPicker-Day--selected):not(.DayPicker-Day--outside):hover {
      background-color: ${rgba(theme.colors.blue, 0.7)};
      color: ${theme.colors.white};
    }

    .DayPickerInput {
      display: inline-block;
      width: 100%;
    }

    .DayPickerInput-OverlayWrapper {
      position: relative;
    }

    .Selectable
      .DayPicker-Day--selected:not(.DayPicker-Day--start):not(.DayPicker-Day--end):not(.DayPicker-Day--outside) {
      background-color: ${theme.colors.blue};
      color: ${theme.colors.white};
    }

    .Selectable .DayPicker-Day {
      border-radius: 50%;
    }
  `}
`
