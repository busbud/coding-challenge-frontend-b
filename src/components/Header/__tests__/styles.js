import styles from '../styles';

const theme = {
  spacing: {
    unit: 8,
  },
};

describe('component | Header | styles', () => {
  it('should be defined', () => {
    // given
    const css = styles(theme);

    // then
    expect(css).toBeDefined();
  });

  it('should match snapshot', () => {
    // given
    const css = styles(theme);

    // then
    expect(css).toMatchSnapshot();
  });

  it('should have proper rules according to provided theme', () => {
    // given
    const css = styles(theme);

    // then
    expect(css.toolbar.paddingTop).toBe(`${theme.spacing.unit}px`);
    expect(css.toolbar.paddingBottom).toBe(`${theme.spacing.unit}px`);
  });
});
