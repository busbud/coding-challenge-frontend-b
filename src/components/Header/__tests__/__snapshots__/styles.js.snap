// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component | Header | styles should match snapshot 1`] = `
Object {
  "item": Object {
    "display": "flex",
    "height": "100%",
    "justifyContent": "space-between",
  },
  "root": Object {
    "flexGrow": 1,
    "height": "4.4rem",
  },
  "toolbar": Object {
    "height": "100%",
    "padding": 0,
    "paddingBottom": "8px",
    "paddingTop": "8px",
  },
}
`;
